### PR TITLE
rpc: split spork manipulation logic to distinct "sporkupdate" call

### PR DESCRIPTION
**Pull request by kittywhiskers**

### CODE REVIEW

File: src/rpc/misc.cpp [hunk 1]
```diff
@@ -1318,6 +1323,7 @@ static const CRPCCommand commands[] =
     /* Dash features */
     { "dash",               "mnsync",                 &mnsync,                 {} },
     { "dash",               "spork",                  &spork,                  {"arg0","value"} },
```
Comment (UdjinM6) [inline src/rpc/misc.cpp:1325]:
```suggestion
    { "dash",               "spork",                  &spork,                  {"command"} },
```

File: src/rpc/client.cpp [hunk 2]
```diff
@@ -165,6 +165,7 @@ static const CRPCConvertParam vRPCConvertParams[] =
     { "logging", 0, "include" },
     { "logging", 1, "exclude" },
     { "spork", 1, "value" },
```
Comment (UdjinM6) [inline src/rpc/client.cpp:167]:
```suggestion
```

File: src/rpc/misc.cpp [hunk 3]
```diff
@@ -1318,6 +1323,7 @@ static const CRPCCommand commands[] =
...
+    { "dash",               "sporkupdate",            &sporkupdate,            {"arg0","value"} },
```
Comment (UdjinM6) [inline src/rpc/misc.cpp:1326]:
```suggestion
    { "dash",               "sporkupdate",            &sporkupdate,            {"name","value"} },
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**